### PR TITLE
Use recursive autosummary

### PR DESCRIPTION
**Pull request by s-weigand**
`sphinx autosummary` finally introduced and recursice option with [version 3.1 ](https://www.sphinx-doc.org/en/master/usage/extensions/autosummary.html), thus my [`api_documentation hack`](https://github.com/glotaran/pyglotaran/blob/5466097596b2c735f7dbb82064658530c79d4b85/docs/source/generate_api_doc.py) isn't needed anymore, to generate proper API docs.

**Testing**

Passing the tests is mandatory.

### CODE REVIEW

**Comment by s-weigand** [comment]
And to easily compare this a future PR's concerning the docs I added an artifact upload for docs generated by the CI.

File: docs/source/user_documentation/api_documentation.rst [hunk 1]
```diff
@@ -1,89 +1,10 @@
...
-    glotaran.model.model_decorator
-    glotaran.model.model_property
-    glotaran.model.util
-    glotaran.parameter
-    glotaran.parameter.parameter
-    glotaran.parameter.parameter_group
-    glotaran.parse
-    glotaran.parse.parser
-    glotaran.parse.register
+    glotaran
```
Comment (jsnel) [inline docs/source/user_documentation/api_documentation.rst:10]: nice simplification

File: glotaran/builtin/file_formats/ascii/wavelength_time_explicit_file.py [hunk 2]
```diff
@@ -285,7 +285,7 @@ def get_data_file_format(line):
 def read_ascii_time_trace(fname: str, prepare: bool = True) -> xr.Dataset:
     """Reads an ascii file in wavelength- or time-explicit format.
 
-    See [1] for documentation of this format.
+    See [1]_ for documentation of this format.
```
Comment (jsnel) [inline glotaran/builtin/file_formats/ascii/wavelength_time_explicit_file.py:288]: What's the underscore doing here?

**Review by jsnel — CHANGES_REQUESTED** [review]
Thank you for the PR, simplifies things a lot. 

I left one comment for you to check, if that checks out it's ready to be merged.

File: glotaran/builtin/file_formats/ascii/wavelength_time_explicit_file.py (hunk 2 above)
Comment (s-weigand) [reply glotaran/builtin/file_formats/ascii/wavelength_time_explicit_file.py:288]: This is just the proper way to [invoke the citation](https://numpy.org/devdocs/docs/howto_document.html).

https://github.com/glotaran/pyglotaran/blob/3c1c4ae6c8d24e8820144a44d4f3bcc3a0bc5040/glotaran/builtin/file_formats/ascii/wavelength_time_explicit_file.py#L299-L301

One warning less from sphinx 😉

Comment (jsnel) [reply glotaran/builtin/file_formats/ascii/wavelength_time_explicit_file.py:288]: Ok, thanks!

**Pull request merged**